dashboard/app/models/user.rb

```diff
@@ -1055,7 +1055,7 @@ def next_unpassed_visible_progression_level(script)
     # will be redirected to /congrats.
     return nil if completed?(script)
 
-    visible_sls = visible_script_levels(script).reject {|sl| sl.bonus || sl.level.unplugged? || sl.locked?(self)}
+    visible_sls = visible_script_levels(script).reject {|sl| sl.bonus || sl.level.unplugged?}
```

Comment:
This was breaking tests in users_test_helper and stage_lock_test.

Reply:
Can we pick a better name for this variable? Even something more generic like `potential_sls` would get away from the implication that all we care about is "visible," when there are actually more conditions.

Reply:
Fair point.  I started only thinking about visibility and ended up here 🙃.  I updated visible to valid and explain in the comment that valid refers to levels that are visible, not unplugged and not bonus. Better?

Reply:
:+1: Way better!